src/app/add-data-set-dialog/add-data-set-dialog.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+
+import { HttpHeaders } from '@angular/common/http';
+
+import { RequestService } from '../request/request.service'
+import { MatDialogRef} from "@angular/material/dialog";
+
+interface Project {
+  name: string;
+  url: string;
+}
+
+@Component({
+  selector: 'app-add-data-set-dialog',
+  templateUrl: './add-data-set-dialog.component.html',
+  styleUrls: ['./add-data-set-dialog.component.css']
+})
+
+export class AddDataSetDialogComponent implements OnInit {
+
+  projects: Project[] = [{name: '', url: ''}];
+
+  constructor(private requestService: RequestService, private dialogRef: MatDialogRef<AddDataSetDialogComponent>) { }
+
+  public addEmptyProject() {
+    this.projects.push({name: '', url: ''});
+  }
+
+  public createDataSets(){
+    if (this.isValidInput()) {
+      let header = {
+        headers: new HttpHeaders()
+          .set('Content-Type', 'application/json')
+      }
+      this.requestService.sendRequest('POST', 'dataset', this.projects, header.headers).subscribe(res => this.dialogRef.close(res));
```

Comment:
Usually the point of the service is to encapsulate all HTTP related stuff so that the component does not need to know whether something is a post request, what the headers are etc.

Essentially, you would want a DatasetService on which you can invoke createDataSet(this.projects). That way the component does not care that an HTTP request is made. By centralizing HTTP processing to the service, you don't have a bunch of components that are familiar with HTTP.